sentry-core/src/main/java/io/sentry/transport/HttpTransport.java

```diff
@@ -0,0 +1,181 @@
+package io.sentry.transport;
+
+import static io.sentry.SentryLevel.DEBUG;
+import static io.sentry.SentryLevel.ERROR;
+
+import io.sentry.ISerializer;
+import io.sentry.SentryEvent;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.util.Nullable;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.Proxy;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.function.Consumer;
```

Comment:
Added in API level 24 Consumer

Reply:
How does one find this out without checking every bloody method/class in all of `java` package? :)

Reply:
static code analysis is the way to go (eg lint), but this is a java library and it doesn't seem to check `NewApi` as @ninniuz pointed out. might be needed to create a custom lint rule, I can look into it

Reply:
![Screenshot 2019-10-09 at 13 52 55](https://user-images.githubusercontent.com/5731772/66479170-3e263580-ea9c-11e9-9442-bddd9b827bea.png)
you may use IntelliJ for now till we find a way to break the build based on those warnings.

Reply:
We need to solve this. Worst case we just make `sentry-core` an android library for now. Or at least one build configuration for CI that builds it as an android library

Reply:
Huh, for reasons unbeknownst to me, I had Android support disabled in Intellij. I now see the squiggly lines! :)

Reply:
@metlos just submitted a PR https://github.com/getsentry/sentry-android/pull/23 so after merging, build will fail when using APIs which is not retro compatible

Reply:
Actually not, because the lint doesn't support the Android API checks for java libraries.

Reply:
Ah, I see what you did there :) the `checkDependencies` will cause the android part to fail if core contains API errors?

Reply:
yep, so we are safe to write code which is compatible with android API 14 and if not, build will fail